tests/capture/input_generators/test_attack_point_iterator.py

```diff
@@ -145,3 +146,213 @@ def test_attack_point_iterator_unrestricted_generator_all_args_len():
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(iter(build_attack_points_iterator(config)))
+    assert output == list(
+        iter(build_attack_points_iterator(config["configuration"])))
+
+
+def test_attack_point_iterator_repeat_one_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == len(config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_two():
+    config = {
+        "operation": "repeat",
+        "repetitions": 2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = list(iter(build_attack_points_iterator(config)))
+    assert output == list(
+        iter(build_attack_points_iterator(
+            config["configuration"]))) * config["repetitions"]
+
+
+def test_attack_point_iterator_repeat_two_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 2,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == config["repetitions"] * len(
+        config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_three_len():
+    config = {
+        "operation": "repeat",
+        "repetitions": 3,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1, 2, 3]
+        }
+    }
+    output = len(build_attack_points_iterator(config))
+    assert output == config["repetitions"] * len(
+        config["configuration"]["values"])
+
+
+def test_attack_point_iterator_repeat_infinite():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": [1]
+        }
+    }
+    output = []
+    i = 0
+    for value in build_attack_points_iterator(config):
+        output.append(value)
+        i += 1
+        if i > 5:
+            break
+
+    assert output == list(
+        iter(build_attack_points_iterator(config["configuration"]))) * i
+
+
+def test_attack_point_iterator_repeat_infinite_no_values():
+    config = {
+        "operation": "repeat",
+        "repetitions": -1,
+        "configuration": {
+            "operation": "constants",
+            "name": "key",
+            "values": []
+        }
+    }
+    output = []
+    i = 0
+    for value in build_attack_points_iterator(config):
+        output.append(value)
+        i += 1
+        if i > 5:
+            break
+
+    assert output == []
```

Comment:
This is good. What is unexpected is that `len(build_attack_points_iterator(config))` raises `LengthIsInfiniteException` even when nothing is iterated.